### PR TITLE
SNOW-9: Show the current menu indicator for posts that have subcategories

### PR DESCRIPTION
**Pull request by grrrero**
JIRA: [SNOW-9](https://issues.fluidproject.org/projects/SNOW/issues/SNOW-9)

To activate the breadcrumbs to the original/main category, we need to add 'current-menu-item' CSS class based on the 'current-post-ancestor' (or 'current-page-ancestor') class of the current page or post. Thus:

```
function make_current_menu_item( $classes, $item ) {
if (in_array('current-post-ancestor', $classes) || in_array('current-page-ancestor', $classes) || in_array('current-menu-item', $classes) ) {
 $classes[] = 'current-menu-item';
 }
 return $classes;
}
add_filter( 'nav_menu_css_class', 'make_current_menu_item', 10, 2 );
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/node_modules/
+Gruntfile.js
+package.json
```
Comment (cindyli) [inline .gitignore]: Add a new line at the end of the file to avoid the github warning.

File: category.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * To make a category landing page:
+ * 1. Make a post.
+ * 2. Give it a name you would recognize like "About Us Landing Page".
+ *    This title is ignored and is useful only to identify it easily in the
+ *    Wordpress admin.
+ * 3. Add the following into the post's Custom Fields:
+ *        name:  is_landing_page
+ *        value: true
+ * 4. Assign the post to the category it's supposed to be a landing page for.
```
Comment (cindyli) [inline category.php]: This sentence is not quite understandable.

File: functions.php [hunk 3]
```diff
@@ -165,43 +127,116 @@ public function widget( $args ) {
 }
 
 function snow_upcoming_workshops() { 
-  register_widget( new snow_panel_widget('snow_upcoming_workshops','Upcoming Workshops','7'));
+  register_widget( new snow_panel_widget('snow_upcoming_workshops','Upcoming Workshops','8'));
```
Comment (cindyli) [inline functions.php]: The parameters to generate snow panels can be defined in a centralized array, such as in a constant or in a global variable so that modifying the widget title or category can be as simple as modifying a centralized array. For example:

```
$snow_panels = array(
    "snow_upcoming_workshops" => array(
        "id": => "snow_upcoming_workshops",
        "title" => "Upcoming Workshops",
        "category" => "8"
    ),
    ...
);

register_widget( new snow_panel_widget($snow_panels["snow_upcoming_workshops"]["id"], $snow_panels["snow_upcoming_workshops"]["titel"], $snow_panels["snow_upcoming_workshops"]["category"]));
```

Even more, the constructor of the class `snow_panel_widget` can be modified to accept an array as a parameter, then the called function can be simplified to:
```
register_widget( new snow_panel_widget($snow_panels["snow_upcoming_workshops"]));
```

The same comment applies to the code at line 13-70 that registers multiple sidebars.

File: functions.php [hunk 4]
```diff
@@ -165,43 +127,116 @@ public function widget( $args ) {
...
-	        foreach($section['widgets'] as $widget) {
-	            if (is_active_sidebar($widget)) {
-	                dynamic_sidebar($widget);
-	            }
-        	}
-        }
-        echo '</'. $el_name . '>';
-    };
+
+if (function_exists('register_sidebar')) {
```
Comment (cindyli) [inline functions.php]: At line 148, 152, the existence of function `register_sidebar` is verified before being called. Why does this not verified for the same function call at line 13-70. Also, is it possible to combine all these call for  `register_sidebar()` into one `if` block?

File: functions.php [hunk 5]
```diff
@@ -165,43 +127,116 @@ public function widget( $args ) {
...
+        );
+        $sidebar_query = new WP_Query($args);
+
+        if ($sidebar_query->have_posts()) :
+
+            /*
+            If there are more than 1 post in the current category,
+            then the first item should be a link to the category.
+
+            Otherwise if there's just 1 post, then it can be ignore as it
```
Comment (cindyli) [inline functions.php]: ignore -> ignored

File: sidebar-page.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * The sidebar containing the main widget area
+ *
+ * @link https://developer.wordpress.org/themes/basics/template-files/#template-partials
+ *
+ * @package a11y
+ */
+
+global $category_landing_id;
```
Comment (cindyli) [inline sidebar-page.php]: This var is defined but never used.

File: sidebar-post.php [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<?php
+/**
+ * The sidebar containing the main widget area
+ *
+ * @link https://developer.wordpress.org/themes/basics/template-files/#template-partials
+ *
+ * @package a11y
+ */
+
+global $category_landing_id;
```
Comment (cindyli) [inline sidebar-post.php]: Same as above, defined but never used.

File: sidebar.php [hunk 8]
```diff
@@ -9,80 +9,3 @@
 
 global $category_landing_id;
```
Comment (cindyli) [inline sidebar.php]: Is this file still needed?

**Comment by cindyli** [comment]
If `style.css` is built from `stylus/style.styl`, `style.css` should be removed to avoid the unsynchronization between these two files in the github. Instead, add a grunt task or add comments in the README to help or indicate the build of `style.css` is necessary before using this theme.

File: widgets.php [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<?php
```
Comment (cindyli) [inline widgets.php]: Shall this empty file be removed?

**Comment by cindyli** [comment]
Shall `stylus/style.styl.orig` be removed?

File: widgets.php (hunk 9 above)
Comment (grrrero) [reply widgets.php]: Removed!

**Comment by grrrero** [comment]
Hi @cindyli I've made the changes based on your review.

File: category.php (hunk 2 above)
Comment (cindyli) [reply category.php]: Please address this comment.

Comment (grrrero) [reply category.php]: @cindyli I'll talk to @jhung as he is the one who originally wrote this. This refers to the Wordpress dashboard for the additional options per post.

File: sidebar-page.php (hunk 6 above)
Comment (cindyli) [reply sidebar-page.php]: Can you address this comment too?

File: sidebar-post.php (hunk 7 above)
Comment (cindyli) [reply sidebar-post.php]: Same here.

File: style.css [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /*
```
Comment (cindyli) [inline style.css]: Shall `style.css` be removed?

File: sidebar-post.php (hunk 7 above)
Comment (grrrero) [reply sidebar-post.php]: @cindyli This is used in the create_sidebar() function in functions.php

File: style.css (hunk 10 above)
Comment (grrrero) [reply style.css]: Yes I'll remove and we'll document it in README :)

Comment (grrrero) [reply style.css]: style.styl.orig as well, that was created unnecessarily.

Comment (grrrero) [reply style.css]: After talking to @cindyli and @jhung we've decided to keep both .styl and .css files because at the moment Wordpress does not have the global capacity to compile stylus files into css and thus we would either need to create a Grunt task (which would be time consuming) or compile it manually locally and then push it and make sure the two files are synchronised.

See JIRA here: https://issues.fluidproject.org/projects/SNOW/issues/SNOW-14

Comment (grrrero) [reply style.css]: After talking to @cindyli and @jhung we've decided to keep both .styl and .css files because at the moment Wordpress does not have the global capacity to compile stylus files into css and thus we would either need to create a Grunt task (which would be time consuming) or compile it manually locally and then push it and make sure the two files are synchronised.

See JIRA here: https://issues.fluidproject.org/projects/SNOW/issues/SNOW-14

File: category.php (hunk 2 above)
Comment (grrrero) [reply category.php]: Removed this file after talking to @jhung; no changes were made to override in the parent theme.

File: functions.php (hunk 5 above)
Comment (grrrero) [reply functions.php]: Fixed this.

File: sidebar-page.php (hunk 6 above)
Comment (grrrero) [reply sidebar-page.php]: It's used in create_sidebar() in functions.php and configured via the WP dashboard. Also used by the a11y parent theme.

**Comment by cindyli** [comment]
This pull request looks good and can be merged into the master.

**Comment by grrrero** [comment]
Thank you so much @cindyli!!!
